views/default/resources/friend_request.php

```diff
@@ -1,12 +1,17 @@
 <?php
 
-elgg_gatekeeper();
-
-$username = elgg_extract('username', $vars);
-$user = get_user_by_username($username);
+$request = elgg_extract('request', $vars);
+/* @var $request \Elgg\Request */
+
+$username = $request->getParam('username');
+if ($username) {
+	$user = get_user_by_username($username);
+} else {
+	$user = elgg_get_logged_in_user_entity();
+}
 
-if (!$user instanceof ElggUser || !$user->canEdit()) {
-	forward('', '404');
+if (!$user) {
```

Comment:
you removed the check `$user->canEdit()` this would allow users to see friends requests of other users, not something we want.